src/relay/transforms/annotate_texture_storage.cc

```diff
@@ -479,7 +584,7 @@ class CollectVirtualDevices : public transform::DeviceAwareExprVisitor {
  * \param expr The expression.
  * \return The device based storage mapping.
  */
-Map<Expr, Array<String>> CollectStorageInfo(const Expr& expr) {
+Map<Expr, Map<Expr, Array<String>>> CollectStorageInfo(const Expr& expr) {
```

Comment:
we need to describe somewhere in this file/classes/methods why we introduced new map in existing map.